validator/rpc/standard_api.go

```diff
@@ -693,3 +695,41 @@ func validatePublicKey(pubkey []byte) error {
 	}
 	return nil
 }
+
+// SetVoluntaryExit TBD
+func (s *Server) SetVoluntaryExit(ctx context.Context, req *ethpbservice.SetVoluntaryExitRequest) (*ethpbservice.SetVoluntaryExitResponse, error) {
+	if s.validatorService == nil {
+		return nil, status.Error(codes.FailedPrecondition, "Validator service not ready")
+	}
+	if err := validatePublicKey(req.Pubkey); err != nil {
+		return nil, status.Error(codes.FailedPrecondition, err.Error())
+	}
+	if s.wallet == nil {
+		return nil, status.Error(codes.FailedPrecondition, "No wallet found")
+	}
+	km, err := s.validatorService.Keymanager()
+	if err != nil {
+		return nil, err
+	}
+	sve, err := client.CreateSignedVoluntaryExit(
+		ctx,
+		s.beaconNodeValidatorClient,
+		s.beaconNodeClient,
+		km.Sign,
+		req.Pubkey,
+		req.Epoch,
+	)
+	if err != nil {
+		return nil, status.Error(codes.FailedPrecondition, "Could not create voluntary exit")
+	}
+
+	return &ethpbservice.SetVoluntaryExitResponse{
+		Data: &ethpbservice.SetVoluntaryExitResponse_SignedVoluntaryExit{
+			Message: &ethpbservice.SetVoluntaryExitResponse_SignedVoluntaryExit_VoluntaryExit{
+				Epoch:          strconv.FormatUint(uint64(sve.Exit.Epoch), 10),
```

Comment:
do you need all these conversions here?

Reply:
It was the only way I was able to return a string, I am open to suggestions if there is a better way.

Reply:
Just saw Radek's comment suggesting a better way